src/ipa_pc/mod.rs

```diff
@@ -325,12 +348,6 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
         supported_degree: usize,
         p: &LabeledPolynomial<G::ScalarField>,
     ) -> Result<(), Error> {
-        if p.degree() > supported_degree {
-            return Err(Error::TooManyCoefficients {
-                num_coefficients: p.degree() + 1,
-                num_powers: supported_degree + 1,
-            });
-        }
 
         if let Some(bound) = p.degree_bound() {
             if bound < p.degree() || bound > supported_degree {
```

Comment:
Using segmentation, the check that the degree bound must not exceed the supported_degree is not needed.

Reply:
Ok, this function is removed now

Reply:
As understand the degree bound still cannot be lower then polynomial degree, so I have returned the function, but with the only check: bound < p.degree()